src/libultra/os/initialize.c

```diff
@@ -89,19 +91,21 @@ void OSINITIALIZE_FUNC(void) {
         osViClock = VI_NTSC_CLOCK;
     }
 
+#if (LIBULTRA_MAJOR_VERSION == LIBULTRA_VERSION_I && LIBULTRA_MINOR_VERSION == 1) || (LIBULTRA_MAJOR_VERSION > LIBULTRA_VERSION_I)
     // If PreNMI is pending, loop until reset
     if (__osGetCause() & CAUSE_IP5) {
         while (true) {
             ;
         }
     }
+#endif
 
     IO_WRITE(AI_CONTROL_REG, AI_CONTROL_DMA_ON);
     IO_WRITE(AI_DACRATE_REG, AI_MAX_DAC_RATE - 1);
     IO_WRITE(AI_BITRATE_REG, AI_MAX_BIT_RATE - 1);
 }
 
-#if PLATFORM_N64
+#if LIBULTRA_MAJOR_VERSION == LIBULTRA_VERSION_I && LIBULTRA_MINOR_VERSION == 1
```

Comment:
what about doing something like
```
ifeq ($(PLATFORM),N64)
$(BUILD_DIR)/src/libultra/os/%.o: LIBULTRA_VERSION := J
endif
```
in the Makefile instead? I think this is more natural if we consider OoT's libultra to be "a mix of I and J" rather than "I revision 1". I'm not sure which way of thinking about it is better though

Reply:
We know that a patch for version I did exist, "2.0I_patchNOA9801", so the approach I took feels to me like it would be the sort of approach ultralib would take rather than building a subset files with more recent versions.

Reply:
I see, alright. To bikeshed the names a bit, what about `LIBULTRA_MAJOR_VERSION` -> `LIBULTRA_VERSION` and `LIBULTRA_MINOR_VERSION` -> `LIBULTRA_PATCH`? The current names are rather long, and maybe this is more consistent with the name `LIBULTRA_VERSION_I` anyway

Reply:
I originally planned to call them that but `LIBULTRA_VERSION` in `rom_header.h` kind of disrupted that. Do you have any suggestions on what I can rename that to in order to make room for these?

Reply:
good question, using the terminology of https://n64brew.dev/wiki/ROM_Header maybe

```
LIBULTRA_VERSION(2, 0, LIBULTRA_REVISION)
```
with `LIBULTRA_REVISION`/`LIBULTRA_PATCH` and e.g. `LIBULTRA_REVISION_I`.

Also could we cleverly write the ASCII character as `.byte 'A' + revision` instead of having a separate define for `LIBULTRA_VERSION_ID`?

Reply:
I think that should be possible, I'll try it out

Reply:
I couldn't really do some of this in the end since it'd change how `rom_header.h` works pretty significantly. Specifically, I don't really see a nice way to allow users to write `OS_VERSION(2, 0, L)` while also allowing `OS_VERSION(2, 0, LIBULTRA_VERSION)`.

Reply:
What `OS_VERSION(2, 0, 'L')` with `#define LIBULTRA_VERSION_L 0x4C`?

Reply:
Yeah that works, done

Reply:
huh I didn't realize CPP could compare character literals, neat